tests/test_date.py

```diff
@@ -118,163 +191,225 @@ def test_should_one_date_for_each_month(self, low, high, expected_results):
-        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25)),
-    ])
-    def test_empty_period(self, low, high):
-        results = list(date.get_intersecting_periods(low, high, period='month'))
-        self.assertEquals([], results)
+    def given_period_low(self, low):
+        self.period_low = low
+
+    def given_period_high(self, high):
+        self.period_high = high
+
+    def given_period(self, period):
+        self.period = period
+
+    def when_intersecting_period_calculated(self):
+        self.intersected_dates = list(date.get_intersecting_periods(
+            self.period_low, self.period_high, period=self.period))
+
+    def then_should_one_date_for_each_date(self):
+        self.assertEquals(self.expected_results, self.intersected_dates)
+
+    def then_date_range_length_is(self, size):
+        self.assertEquals(size, len(self.intersected_dates))
 
+    def then_all_dates_are_present_in_range(self):
+        self.assertEquals(self.period_low, self.intersected_dates[0])
 
-class ParseDateWithFormats(unittest.TestCase):
+    def then_period_is_empty(self):
+        self.assertEquals([], self.intersected_dates)
 
-    def test_shouldnt_parse_invalid_date(self):
-        self.assertIsNone(date.parse_with_formats('yesterday', ['%Y-%m-%d'])['date_obj'])
+    def when_multiple_periods_are_parsed(self, periods):
+        self.results = []
+        for period in periods:
+            try:
+                result = date.get_intersecting_periods(self.period_low, self.period_high, period=period).next()
+            except Exception as error:
+                result = error
+            finally:
+                self.results.append(result)
+
+    def then_should_reject_easily_mistaken_dateutil_arguments(self, periods):
+        for p in izip(self.results, periods):
+            self.assertIsInstance(p[0], ValueError)
+            self.assertEqual('Invalid period: {}'.format(p[1]), p[0].message)
+
+
+class ParseDateWithFormats(BaseTestCase):
+    def setUp(self):
+        super(ParseDateWithFormats, self).setUp()
+        self.dt_data = NotImplemented
+
+    def test_date_with_not_matching_format_is_not_parsed(self):
+        self.given_invalid_date('yesterday')
+        self.then_shouldnt_parse_invalid_date()
 
     def test_should_parse_date(self):
-        result = date.parse_with_formats('25-03-14', ['%d-%m-%y'])
-        self.assertEquals(datetime(2014, 3, 25).date(), result['date_obj'].date())
+        self.given_a_date(datetime(2014, 3, 25))
+        self.when_date_is_parsed_with_format('25-03-14', '%d-%m-%y')
+        self.then_parsed_date_is()
 
     def test_should_use_current_year_for_dates_without_year(self):
-        today = datetime.today()
+        self.given_current_year()
+        self.given_day(16)
+        self.given_month(9)
+        self.when_date_is_parsed_with_format('09.16', "%m.%d")
+        self.then_should_use_current_year_for_dates_without_year()
```

Comment:
Because that method executes ```self.dt_data = date.parse_with_formats(some_date, [some_format])``` and ```self.dt_data`` is used in ```then_parsed_date_is```. Now it's my assumption that the when function is passing it's result to the then statement. I guess that the when does not have to change here but it is my thinking that you would want to validate the result of the when statement.